server-spi-private/src/main/java/org/keycloak/component/ComponentModelScope.java

```diff
@@ -116,7 +116,9 @@ public Scope scope(String... scope) {
 
     @Override
     public Set<String> getPropertyNames() {
-        throw new UnsupportedOperationException("Not implemented");
+        Set<String> keys = this.componentConfig.getConfig().keySet();
+        keys.addAll(this.origScope.getPropertyNames());
+        return keys;
```

Comment:
Thanks for providing an implementation here! Finally a way to iterate over the properties!
(resolve the comment)

Reply:
Just caught my eye - this implementation needs updating: `keys` do not support `addAll`, see https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#keySet--

Reply:
Thanks, I've completely missed that one. Fixed.